server/src/main/java/org/candlepin/controller/RevocationOp.java

```diff
@@ -0,0 +1,171 @@
+/**
+ * Copyright (c) 2009 - 2012 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.controller;
+
+import com.google.inject.persist.Transactional;
+import org.candlepin.common.exceptions.ForbiddenException;
+import org.candlepin.model.Entitlement;
+import org.candlepin.model.Pool;
+import org.candlepin.model.PoolCurator;
+import org.candlepin.policy.EntitlementRefusedException;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+
+/**
+ * Created by wpoteat on 5/11/17.
+ */
+public class RevocationOp {
+    private List<Pool> pools;
+    private Set<Entitlement> entitlementsToRevoke;
+    private Map<Entitlement, Long> entitlementsToAdjust;
+    private Map<Pool, List<Pool>> sharedPools;
+    private PoolCurator poolCurator;
+    private long newConsumed;
+
+    public RevocationOp(PoolCurator poolCurator, List<Pool> pools) {
+        entitlementsToRevoke = new HashSet<Entitlement>();
+        entitlementsToAdjust = new HashMap<Entitlement, Long>();
```

Comment:
if I understood right, there are only shared ents in this list. would it make sense to change its name to sharedEntitlementsToAdjust?